ssl/record/rec_layer_s3.c

```diff
@@ -1066,13 +1094,16 @@ int do_ssl3_write(SSL *s, int type, const unsigned char *buf,
             goto err;
         }
 
+ mac_done:
```

Comment:
Couldn't this be moved further down to just above the SSL3_BUFFER_set_left() call below? You might also have to move the SSL3_RECORD_set_type() call as well.

Reply:
Yes, I've moved it now.